src/pages/loginphone.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState } from 'react'
+import { NextPage } from 'next'
+
+import EnterPasswordPanel from '@/components/Login/EnterPasswordPanel'
+import EnterPhonePanel from '@/components/Login/EnterPhonePanel'
+import LoginHeader from '@/components/Login/LoginHeader'
+import { withPublic } from '@/components/PrivateRoute'
+
+const Login: NextPage = () => {
+  const pageTitle = 'Phone Login'
+  const primaryMessage = 'Enter your phone number'
+  const secondaryMessage = 'Please enter your phone number'
+
+  const [phonenumber, setphonenumber] = useState<string>('')
+  // const [password, setpassword] = useState<string>('')
+  const [panel, setpanel] = useState('phone')
+
+  function handlePhoneSubmit(phoneNumber: string) {
+    setphonenumber(phoneNumber)
+    setpanel('password')
+  }
+
+  function togglePanel() {
+    setpanel('phone')
+  }
+
+  function renderPanel(state: string) {
```

Comment:
Please refactor this to use components instead of functions. (See why [here](https://www.youtube.com/watch?v=QuLfCUh-iwI))